mesh/mesh.cpp

```diff
@@ -7283,28 +7283,24 @@ void Mesh::UniformRefinement(int ref_algo)
    {
       NURBSUniformRefinement();
    }
-   else if (ref_algo == 0 && Dim == 3 && meshgen == 1)
-   {
-      UniformRefinement3D();
-   }
-   else if (meshgen == 1 || ncmesh)
+   else if (ncmesh)
    {
       Array<int> elem_to_refine(GetNE());
       for (int i = 0; i < elem_to_refine.Size(); i++)
       {
          elem_to_refine[i] = i;
       }
-
-      if (Conforming())
-      {
-         // In parallel we should set the default 2nd argument to -3 to indicate
-         // uniform refinement.
-         LocalRefinement(elem_to_refine);
-      }
-      else
+      GeneralRefinement(elem_to_refine, 1);
+   }
+   else if (ref_algo == 1 && meshgen == 1)
+   {
+      // algorithm "B" for an all-tet mesh
```

Comment:
This comment is incorrect -- this branch will be executed in 1D, 2D (all triangle mesh), and 3D (all tet mesh).

Also, I just realized that -- the method, as it is, will crash for any 1D mesh (unless one explicitly uses `ref_algo == 1`).

Reply:
Both should hopefully be fixed now, thanks.